janus_core/cli/phonons.py

```diff
@@ -0,0 +1,263 @@
+        Default is 2.
+    displacement : float
+        Displacement for force constants calculation, in A. Default is 0.01.
+    thermal : bool
+        Whether to calculate thermal properties. Default is False.
+    temp_start : float
+        Start temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 0.0.
+    temp_end : float
+        End temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 1000.0.
+    temp_step : float
+        Temperature step for CV calculations, in K. Unused if `thermal` is False.
+        Default is 50.0.
+    dos : bool
+        Whether to calculate and save the DOS. Default is False.
+    pdos : bool
+        Whether to calculate and save the PDOS. Default is False.
+    minimize : bool
+        Whether to minimize structure before calculations. Default is False.
+    fmax : float
+        Set force convergence criteria for optimizer in units eV/Å.
+        Default is 0.1.
+    minimize_kwargs : Optional[dict[str, Any]]
+        Other keyword arguments to pass to geometry optimizer. Default is {}.
+    arch : Optional[str]
+        MLIP architecture to use for geometry optimization.
+        Default is "mace_mp".
+    device : Optional[str]
+        Device to run model on. Default is "cpu".
+    read_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to ase.io.read. Default is {}.
+    calc_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to the selected calculator. Default is {}.
+    file_prefix : Optional[PathLike]
+        Prefix for output filenames. Default is inferred from structure name, or
+        chemical formula.
+    log : Optional[Path]
+        Path to write logs to. Default is "phonons.log".
+    summary : Path
+        Path to save summary of inputs and start/end time. Default is
+        phonons.yml.
+    config : Path
+        Path to yaml configuration file to define the above options. Default is None.
+    """
+    # Check options from configuration file are all valid
+    check_config(ctx)
+
+    [read_kwargs, calc_kwargs, minimize_kwargs] = parse_typer_dicts(
+        [read_kwargs, calc_kwargs, minimize_kwargs]
+    )
+
+    # Set up single point calculator
+    s_point = SinglePoint(
+        struct_path=struct,
+        struct_name=struct_name,
+        architecture=arch,
+        device=device,
+        read_kwargs=read_kwargs,
+        calc_kwargs=calc_kwargs,
+        log_kwargs={"filename": log, "filemode": "w"},
+    )
+
+    log_kwargs = {"filename": log, "filemode": "a"}
+
+    # Check fmax option not duplicated
+    if "fmax" in minimize_kwargs:
+        raise ValueError("'fmax' must be passed through the --fmax option")
+    minimize_kwargs["fmax"] = fmax
+
+    # Set default supercell
+    if not supercell:
+        supercell = [2, 2, 2]
+
+    # Convert single value to list for supercell matrix
+    if len(supercell) == 1:
+        supercell *= 3
```

Comment:
Ok, I pass in `[[2, 2, 2]]` this is duplicated to `[[2, 2, 2], [2, 2, 2], [2, 2, 2]]`, which has length 3, it would fall over later, but I think the check might be overly simplistic. Not certain we need to care, but just to be aware...

Reply:
It checks that the value after --supercell is an integer at the moment, so you can't pass  `--supercell [2, 2, 2]` or even just `--supercell [2]` (with or without quotes around the [ ] )